pennylane_lightning/core/src/utils/Util.hpp

```diff
@@ -41,6 +41,7 @@ namespace Pennylane::Util {
  * @return constexpr std::complex<T>
  */
 template <class T, class U = T>
+    requires std::integral<U> || std::floating_point<U>
```

Comment:
These restrictions could be applied to the rest of compile-time complex scalars. This will preserve the support for the future use cases. Could you add the required concept to the rest of scalars functions in `Utils`?

Reply:
I don't think this line needs to be added anywhere else --- is there a specific set of arith * complex functions that would think is needed? This seems to be the only real * complex in this file, no?

Reply:
I meant whether we need to use concepts for the rest of scalar methods in this file (and maybe everywhere) to prevent such template issues in the future. For example, we can update `ConstMult` and similar ones as follows:

```
template <class T, class U = T>
    requires (std::integral<T> || std::floating_point<T>) && 
             (std::integral<U> || std::floating_point<U>)
inline static constexpr auto ConstMult(std::complex<U> a, std::complex<T> b)
    -> std::complex<T> {
// ...
}
```

Reply:
Given that the the problem here was U being mapped to `std:complex<A>` or `A`, causing the function call to fail, I think the above suggestion is a little different. The std::complex type already validates the incoming `U/T` types, and will prevent issues with the selection and compilation. I don't think it is necessary for the others in this PR.